test/powershell/Language/Parser/Parsing.Tests.ps1

```diff
@@ -309,4 +309,11 @@ Describe 'Unicode escape sequence parsing' -Tag "CI" {
     ShouldBeParseError '"`u219z"' InvalidUnicodeEscapeSequence 1
     ShouldBeParseError '"`u{219z"' UnicodeEscapeSequenceMissingTerminator 7 # error offset "`u{219>>z<<"
     ShouldBeParseError '"`u{}"' InvalidBracketedUnicodeEscapeSequence 1
+    ShouldBeParseError '"`u' InvalidUnicodeEscapeSequence,TerminatorExpectedAtEndOfString 1,0
+    ShouldBeParseError '"`u123' InvalidUnicodeEscapeSequence,TerminatorExpectedAtEndOfString 1,0
+    ShouldBeParseError '"`u219z' InvalidUnicodeEscapeSequence,TerminatorExpectedAtEndOfString 1,0
+    ShouldBeParseError '"`u{' UnicodeEscapeSequenceMissingTerminator,TerminatorExpectedAtEndOfString 4,0
+    ShouldBeParseError '"`u{1' UnicodeEscapeSequenceMissingTerminator,TerminatorExpectedAtEndOfString 5,0
+    ShouldBeParseError '"`u{123456' UnicodeEscapeSequenceMissingTerminator,TerminatorExpectedAtEndOfString 10,0
+    ShouldBeParseError '"`u{1234567' UnicodeEscapeSequenceMissingTerminator,TerminatorExpectedAtEndOfString 10,0
```

Comment:
Your example generates the following two errors:
```
iex '"`u123}'
iex : At line:1 char:2
+ "`u123}
+  ~~~~~
The Unicode escape sequence is not valid. A valid sequence is of the form `uXXXX where X is a hex digit and there must
be four hex digits.
At line:1 char:1
+ "`u123}
+ ~~~~~~~
The string is missing the terminator: ".
At line:1 char:1
+ iex '"`u123}'
+ ~~~~~~~~~~~~~
    + CategoryInfo          : ParserError: (:) [Invoke-Expression], ParseException
    + FullyQualifiedErrorId : InvalidUnicodeEscapeSequence,Microsoft.PowerShell.Commands.InvokeExpressionCommand
```
Effectively `` "`u1`` tells the tokenizer that the Unicode esc seq is not using the bracketed syntax so if it encounters a non-hex digit before it scans four hex digits, it errors telling you that the esc seq isn't valid and tells you what the correct form is.